.github/workflows/release.yml

```diff
@@ -1,63 +0,0 @@
-name: Release
-
-on:
-  release:
-    types: [published]
-
-jobs:
-  style:
-    name: Check style
-    runs-on: ubuntu-22.04
-    steps:
-    - uses: actions/checkout@v4
-    - uses: actions/setup-python@v5
-      with:
-        python-version: '3.12'
-    - uses: pre-commit/action@v3.0.1
-
-  build_wheels:
-    needs: style
-    strategy:
-      matrix:
-        os: [ubuntu-22.04, windows-2022, macos-15]
-    name: Build wheels (${{ matrix.os }})
-    runs-on: ${{ matrix.os }}
-    steps:
-      - uses: actions/checkout@v4
-      - name: Build wheels and run tests
-        uses: pypa/cibuildwheel@v2.22.0
-      - uses: actions/upload-artifact@v4
-        with:
-          name: artifact-${{ matrix.os }}
-          path: ./wheelhouse/*.whl
-
-  build_sdist:
-    needs: style
-    name: Build source distribution
-    runs-on: ubuntu-22.04
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        name: Install Python
-        with:
-          python-version: 3.12
-      - name: Build sdist
-        run: |
-          python -m pip install build numpy
-          python -m build --sdist
-      - uses: actions/upload-artifact@v4
-        with:
-          name: artifact-sdist
-          path: dist/*.tar.gz
-
-  upload-pypi:
-    name: Upload to PyPI
-    needs: [style, build_wheels, build_sdist]
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/download-artifact@v4
-        with:
-          pattern: artifact-*
-          merge-multiple: true
-          path: dist
-      - uses: pypa/gh-action-pypi-publish@v1.12.3
```

Comment:
This was ~80% duplicated with `tests.yml` so I folded both into a `cibuildwheel.yml`. Don't love the name but maybe good enough. Could use templates or something to deduplicate but it's not as readable or easy to work with.